polaris-react/src/components/Pagination/Pagination.tsx

```diff
@@ -183,7 +177,7 @@ export function Pagination({
     <nav aria-label={navLabel} ref={node} className={styles.Pagination}>
       {previousButtonEvents}
       {nextButtonEvents}
-      <ButtonGroup segmented={!label || polarisSummerEditions2023}>
```

Comment:
Since `polarisSummerEditions2023` flag is always true, ButtonGroup should always be segmented.